### ISSUE TITLE
Update to Python3

### ISSUE THREAD

**Issue by aeslaughter** [idaholab/moose#11100]
## Rationale
Python 2.7 will no longer be developed after 1/1/2020, see [PEP0373](https://www.python.org/dev/peps/pep-0373/#id4).

## Description
All python code in MOOSE should be migrated to python 3.

## Impact
This will make python3 a requirement for MOOSE infrastructure items (e.g., TestHarness).

**Comment by permcody** [comment]
That's still a few years off. We'll need to continue support Python 2 as long as the newer versions are not supplied on our minimum systems (Perhaps RHEL).

**Comment by friedmud** [comment]
Yep - still currently against this.

**Comment by rwcarlsen** [comment]
It isn't too hard to write python code that works with both 2 and 3.

**Comment by amjokisaari** [comment]
Conversely, as more software moves to python 3, continuing to use python 2 for MOOSE is going to require more and more users to handle multiple python environments.  It's totally doable but it will start putting up a barrier to entry.  Maybe a note should be added to the installation page about how to do this if necessary?

**Comment by mangerij** [comment]
It is necessary!

This whole having some arbitrary miniconda environment installed parallel
to my main conda environment always wigs me out when I'm updating or
installing new libraries. In fact, I've been blocked on a particular MOOSE
project because I've been wary of reinstalling the MOOSE environment (it is
required for an update added in Oct of last year) for fear that I mess up
all my main python libraries that I use for all sorts of stuff. I'm not wiz
at managing all this crap but it would be nice to have something to follow
like the very clear surefire instructions of installing MOOSE/libMesh
itself.

**Comment by mangerij** [comment]
but i should mention that updating the python3 would probably make the
environment issue even more complicated

**Comment by friedmud** [comment]
@mangerij we never modify your system... including your system python setup... so there is nothing to fear.

You can always remove our environment by simply not sourcing it in your bash_profile.

**Comment by aeslaughter** [comment]
The update is complete; python 2 support will be removed and can reference the closed issue.
